OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -235,24 +235,38 @@ public String getEmail() {
         return activityLoginBinding.emailEt.getText().toString().trim();
     }
 
-    private static final int RESET_PASSWORD_REQUEST_CODE = 0;
-
     private void showResetPasswordDialog() {
-        startActivityForResult(ResetPasswordActivity.newIntent(getEmail()), RESET_PASSWORD_REQUEST_CODE);
+        resetPasswordAlert = ResetPasswordAlertDialogFragment.newInstance(this, getEmail());
+        resetPasswordAlert.setListener(this);
+        resetPasswordAlert.show(getSupportFragmentManager(), null);
     }
 
-    public void showEulaDialog() {
-        environment.getRouter().showWebViewDialog(this, getString(R.string.eula_file_link), getString(R.string.end_user_title));
+    @Override
+    public void onResult(boolean success, @Nullable String errorMessage) {
+        if (success) {
+            if (resetPasswordAlert!=null) {
+                resetPasswordAlert.dismiss();
+            }
+            showErrorDialog(getString(R.string.success_dialog_title_help), getString(R.string.success_dialog_message_help));
+        }
+        else if (resetPasswordAlert!=null) {
+            if (errorMessage!=null) {
+                resetPasswordAlert.showError(errorMessage);
+            }
+            else {
+                // this shouldn't happen, but just in case we are here, dismiss the dialog
+                resetPasswordAlert.dismiss();
+            }
+        }
     }
 
-    public void showNoNetworkDialog() {
-        Bundle args = new Bundle();
-        args.putString(SimpleAlertDialog.EXTRA_TITLE, getString(R.string.reset_no_network_title));
-        args.putString(SimpleAlertDialog.EXTRA_MESSAGE, getString(R.string.reset_no_network_message));
+    @Override
+    public void omDismissed() {
+        resetPasswordAlert=null;
+    }
 
-        SimpleAlertDialog noNetworkFragment = SimpleAlertDialog.newInstance(args);
-        noNetworkFragment.setStyle(DialogFragment.STYLE_NO_TITLE, 0);
-        noNetworkFragment.show(getSupportFragmentManager(), "dialog");
+    public void showEulaDialog() {
```

Comment:
We aren't actually opening a dialog now, so, this function's name and all the related activity, fragment and router function names should be refactored accordingly.